sql3/parser/parser.go

```diff
@@ -355,10 +375,97 @@ func (p *Parser) parseDropStatement() (Statement, error) {
 	case FUNCTION:
 		return p.parseDropFunctionStatement(pos)
 	default:
-		return nil, p.errorExpected(pos, tok, "TABLE, VIEW or FUNCTION")
+		return nil, p.errorExpected(pos, tok, "DATABASE, TABLE, VIEW or FUNCTION")
+	}
+}
+
+func (p *Parser) parseCreateDatabaseStatement(createPos Pos) (_ *CreateDatabaseStatement, err error) {
+	assert(p.peek() == DATABASE)
+
+	var stmt CreateDatabaseStatement
+	stmt.Create = createPos
+	stmt.Database, _, _ = p.scan()
+
+	// Parse optional "IF NOT EXISTS".
+	if p.peek() == IF {
+		stmt.If, _, _ = p.scan()
+
+		pos, tok, _ := p.scan()
+		if tok != NOT {
+			return &stmt, p.errorExpected(pos, tok, "NOT")
+		}
+		stmt.IfNot = pos
+
+		pos, tok, _ = p.scan()
+		if tok != EXISTS {
+			return &stmt, p.errorExpected(pos, tok, "EXISTS")
+		}
+		stmt.IfNotExists = pos
+	}
+
+	if stmt.Name, err = p.parseIdent("database name"); err != nil {
+		return &stmt, err
+	}
+
```

Comment:
This is definitely not a hill I would die on, but with that said, I'm not familiar with `CREATE DATABASE` statements using `WITH`. Where do we use `WITH` that would make it consistent? In `BULK INSERT`, right? But not in `CREATE TABLE` unless I'm missing something.

Also, I was loosely following @gthrone25's [document](https://molecula.atlassian.net/browse/CLOUD-1330).

Reply:
There is no standard for this, so we can do what we want. We use WITH in BULK INSERT, I think we should use WITH in create table, but don't (yet). I used WITH in CREATE MODEL (in another branch). All I am really saying is let's not miss an opportunity to get consistent on syntax, cause once it's out, it's really hard to change.

BTW - that ticket should probably be an Argon ticket in the SQL3 epic or at the least link to an Argon ticket.

Reply:
Agreed there's no one syntax to rule them all here. Consistency is my number one priority, so @paddyjok  has the bigger picture of where things are moving to align. If we are going to add WITH table, it makes sense to add it here. Out of curiosity Pat, are you generally following SQL server for everything? I generally look across multiple DBs, but it'd help to know if that's your main reference.

Reply:
Bit of SQL Server, bit of SQLLite, bit of PostgreSQL, bit of MySQL. I've be preferring to be more explicit than less and to remove some of the noise syntax where it makes sense (e.g SELECT [[ALL]|DISTINCT] doesn't need ALL, but we should support noise AS for aliases because people expect it). WITH falls into the 'more explicit' category. Another one TOP (instead of LIMIT) was in the explicit category because I didn't like that we might end up with TOP(N)/TOPK and LIMIT at the same time and that'd get weird. I think, knowing what we know now, and observing our (albeit) limited user pop. we should deep-six TOP in favor of LIMIT.

Let me model some of the current and upcoming syntax in ebnf/html and I will circulate for comment (we can re-use for doc) so we can decide and move on.

Reply:
+1 on `LIMIT` over `TOP`, but that's just because it's been 25 years since I've used SQL Server and `LIMIT` is burned into my brain from other databases. It also seems to align well with the `LIMIT, OFFSET` pattern should we ever decide to support `OFFSET`.

I have no problem with `WITH`. I'm not used to it, but being explicit seems useful and I like that it designates the start of an options list.

Reply:
+1 in LIMIT and WITH. I always found  the way the options were tacked on to CREATE TABLE a bit unsettling